frontend/src/routes/RouteMain.tsx

```diff
@@ -1,13 +1,22 @@
 import React from 'react';
-import {Route, Routes} from "react-router-dom";
-import {IRoute, mainRoutes} from "./routesConfig";
+import {Navigate, Route, Routes} from "react-router-dom";
+import {IRoute, mainClientRoutes, mainMerchantRoutes} from "./routesConfig";
+import {Paths} from "../common/constansts/paths";
+import {useAppSelector} from "../stores/redux/hooks";
+import {selectLoggedUser} from "../stores/redux/loggedUser/loggedUserSlice";
+import {UserType} from "../types/user";
 
 const RouteMain: React.FC = () => {
+    const loggedUser = useAppSelector(selectLoggedUser);
     return (
         <Routes>
-            {mainRoutes.map((route: IRoute) => (
+            {loggedUser.userType === UserType.Customer ? mainClientRoutes.map((route: IRoute) => (
                 <Route key={route.path} path={route.path} element={route.element}/>
-            ))}
+            )) : mainMerchantRoutes.map((route: IRoute) => (
+                <Route key={route.path} path={route.path} element={route.element}/>
+            ))
+            }
```

Comment:
1) it's impossible to read. 
2) you duplicate the code

Create smth like this:
```typescript
const getInitialState = (loggedUser: ILoggedUser) => loggedUser.userType === UserType.Customer ? mainClientRoutes : mainMerchantRoutes;

const RouteMain: React.FC = () => {
    const loggedUser = useAppSelector(selectLoggedUser); // instead of loggedUser get only userType. This method of getting only type you can move to the slice, as you are going to reuse it in other places
    const [mainRoutes, setMainRoutes] = useState<IRoute[]>(getInitialState(loggedUser));
    
    return (
        <Routes>
            {mainRoutes.map((route: IRoute) => 
                <Route key={route.path} path={route.path} element={route.element}/>)}
            <Route path={Paths.ANY} element={<Navigate to={Paths.Home} replace={true}/>}/>
        </Routes>
    );
}

export default RouteMain;
``` 

same for other cases across the app